docs/DeveloperGuide.adoc

```diff
@@ -98,18 +98,19 @@ image::LogicClassDiagram.png[]
 *API* :
 link:{repoURL}/src/main/java/seedu/address/logic/Logic.java[`Logic.java`]
 
-.  `Logic` uses the `AddressBookParser` class to parse the user command.
+.  `Logic` uses either the `NormalParser` or `InterviewParser` class to parse the user command, depending on the app phase.
 .  This results in a `Command` object which is executed by the `LogicManager`.
-.  The command execution can affect the `Model` (e.g. adding a person).
+.  The command execution can affect the `Model` (e.g. adding an attribute).
 .  The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
-.  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
+.  In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying the
+list of the current attribute.
 
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("add attribute leadership")` API call.
 
-.Interactions Inside the Logic Component for the `delete 1` Command
-image::DeleteSequenceDiagram.png[]
+.Interactions Inside the Logic Component for the `add attribute leadership` Command
+image::AddSequenceDiagram.png[]
```

Comment:
AddCommandParser should parse("attribute leadership") instead, since NormalParser has already parsed it into an Add command.

Reply:
Additionally, the Model.getAttributeList().add("leadership") violates the Law of Demeter. The Model is peeking into the methods of attributes, but LoD aims to prevent objects navigating internal structures of other objects.